src/main/java/uk/gov/hmcts/ccd/domain/service/common/CompoundAccessControlService.java

```diff
@@ -133,7 +133,9 @@ private boolean isCurrentNodeOrAnyChildNodeDeletedWithoutAccess(final JsonNode n
         if (correspondingNewNode.isPresent()) {
             JsonNode newNode = correspondingNewNode.get();
             for (CaseField field : caseField.getFieldType().getCollectionFieldType().getComplexFields()) {
-                if (field.isCompoundFieldType() && existingNode.get(VALUE) != null && existingNode.get(VALUE).get(field.getId()) != null
+                if (field.isCompoundFieldType()
+                    && existingNode.get(VALUE) != null && existingNode.get(VALUE).get(field.getId()) != null
+                    && newNode.get(VALUE) != null && newNode.get(VALUE).get(field.getId()) != null
```

Comment:
This if getting too complicated. Let's move these null checks into a separate private method:
```if (field.isCompoundFieldType() && notNulls(existingNode, newNode, field.getId()) && ....)```